custom/icds_reports/models/aggregate.py

```diff
@@ -1688,3 +1690,51 @@ class Meta(object):
 
     _agg_helper_cls = AggServiceDeliveryReportHelper
     _agg_atomic = False
+
+
+class AggBiharAPIDemographics(models.Model, AggregateMixin):
+    state_id = models.TextField(null=True)
+    district_id = models.TextField(null=True)
+    block_id = models.TextField(null=True)
+    supervisor_id = models.TextField(null=True)
+    awc_id = models.TextField(null=True)
+    month = models.DateField()
+    household_id = models.TextField(null=True)
+    household_name = models.TextField(null=True)
+    hh_reg_date = models.TextField(null=True)
+    hh_num = models.IntegerField(null=True)
+    hh_gps_location = models.TextField(null=True)
+    hh_caste = models.TextField(null=True)
+    hh_bpl_apl = models.TextField(null=True)
+    hh_minority = models.SmallIntegerField(null=True)
+    hh_religion = models.TextField(null=True)
+    hh_member_number = models.IntegerField(null=True)
+    person_id = models.TextField(primary_key=True)
+    person_name = models.TextField(null=True)
+    has_adhaar = models.SmallIntegerField(null=True)
+    bank_account_number = models.TextField(null=True)
+    ifsc_code = models.TextField(null=True)
+    age_at_reg = models.SmallIntegerField(null=True)
+    dob = models.DateField(null=True)
+    gender = models.TextField(null=True)
+    blood_group = models.TextField(null=True)
+    disabled = models.SmallIntegerField(null=True)
+    disability_type = models.TextField(null=True)
+    referral_status = models.TextField(null=True)
+    migration_status = models.SmallIntegerField(null=True)
+    resident = models.SmallIntegerField(null=True)
+    registered_status = models.SmallIntegerField(null=True)
+    rch_id = models.TextField(null=True)
+    mcts_id = models.TextField(null=True)
+    phone_number = models.TextField(null=True)
+    date_death = models.DateField(null=True)
+    site_death = models.TextField(null=True)
+    closed_on = models.DateField(null=True)
+    reason_closure = models.TextField(null=True)
+
+    class Meta(object):
+        db_table = AGG_BIHAR_API_DEMOGRAPHICS
+        unique_together = ('month', 'state_id', 'district_id', 'block_id', 'supervisor_id', 'person_id')  # pkey
```

Comment:
This table is one row per person per month forever? This will be HUGE. Do you have a sense of the space requirements for this?

Reply:
Is there any way we can delete old data in this table?

Reply:
Hey, Yes,  it will be per person per month, But it wont be all person cases forever. it will be only not closed and no migrated cases till that month. 
Space requirement:  I think the space required for one month would be a few GB

Yes, Nitish told we can drop data older than 2 months. But I think we can't implement that right now because In the first time pull they also need Jan data.
I think after first pull we can implement that.

Do you think month wise partition would help here?

Reply:
done the month wise partitioning

Reply:
Yeah, but one row per open person case per month for the duration of the project is still huge. Its many times larger than our already largest table (child health monthly)